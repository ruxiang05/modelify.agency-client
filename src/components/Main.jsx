import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import JobList from './JobList';
import Models from './Models';
import AddJob from './AddJob/AddJob';
import Chat from './Chat';
import Profile from './Profile/Profile';
import NotFound from './NotFound';
import Signup from './Signup';
import Login from './Login';
import AddModel from './AddModel';
import ProtectedRoute from './ProtectedRoute';

const Main = () => (
  <Switch>
    <Redirect from="/" exact to="/jobs" />
    <Route path="/signup" component={Signup} />
    <Route path="/login" component={Login} />
    <ProtectedRoute path="/jobs" exact component={JobList} />
    <ProtectedRoute path="/models" component={Models} />
    <ProtectedRoute path="/jobs/new" component={AddJob} />
    <ProtectedRoute path="/chat" component={Chat} />
    <ProtectedRoute path="/profile" component={Profile} />
    <ProtectedRoute path="/add-model" component={AddModel} />
    <Route component={NotFound} />
  </Switch>
);

export default Main;
