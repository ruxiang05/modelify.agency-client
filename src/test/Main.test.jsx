import React from 'react';
import { shallow } from 'enzyme';
import Main from '../components/Main';

const user = {
  role: 'agent',
};

describe('Main', () => {
  it('should render Main', () => {
    const wrapper = shallow(<Main user={user} />);
    expect(wrapper).toMatchSnapshot();
  });
});
