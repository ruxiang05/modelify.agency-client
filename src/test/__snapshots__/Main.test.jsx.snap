// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render Main 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    user={
      Object {
        "role": "agent",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Redirect
          exact={true}
          from="/"
          push={false}
          to="/jobs"
        />,
        <Route
          component={[Function]}
          path="/signup"
        />,
        <Route
          component={[Function]}
          path="/login"
        />,
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/jobs"
        />,
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/models"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/jobs/new"
        />,
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/chats"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/profile"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/add-model"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/jobs/:id"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/models/:id"
        />,
        <ProtectedRoute
          component={[Function]}
          path="/chats/:id"
        />,
        <Route
          component={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "from": "/",
          "push": false,
          "to": "/jobs",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/jobs",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/models",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/jobs/new",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/chats",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/profile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/add-model",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/jobs/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/models/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "path": "/chats/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Redirect
            exact={true}
            from="/"
            push={false}
            to="/jobs"
          />,
          <Route
            component={[Function]}
            path="/signup"
          />,
          <Route
            component={[Function]}
            path="/login"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/jobs"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/models"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/jobs/new"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/chats"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/profile"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/add-model"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/jobs/:id"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/models/:id"
          />,
          <ProtectedRoute
            component={[Function]}
            path="/chats/:id"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "from": "/",
            "push": false,
            "to": "/jobs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/jobs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/models",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/jobs/new",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/chats",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/add-model",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/jobs/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/models/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "path": "/chats/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
